src/Runtime/jni/CMakeLists.txt

```diff
@@ -1,20 +1,57 @@
 # SPDX-License-Identifier: Apache-2.0
 
 if (ONNX_MLIR_ENABLE_JNI)
+
+  # We use Jsoniter for JNI backend tests. Jackson is another option.
+  # But tests show that Jsoniter is faster and uses less memory even
+  # though it hasn't been updated since 2018.
+  set(JSONITER_REPO_URL https://github.com/json-iterator/java.git)
+  set(JSONITER_REPO_DIR ${CMAKE_CURRENT_BINARY_DIR}/jsoniter)
+  set(JSONITER_VER 0.9.23)
+  set(JSONITER_REPO_JAR ${JSONITER_REPO_DIR}/target/jsoniter-${JSONITER_VER}.jar)
+
+  # Build Jsoniter if an installed version is not found
+  unset(JSONITER_JAR CACHE)
+  find_jar(JSONITER_JAR jsoniter-${JSONITER_VER})
+  if (${JSONITER_JAR} STREQUAL "JSONITER_JAR-NOTFOUND")
+    set(JSONITER_JAR ${JSONITER_REPO_JAR} CACHE STRING "" FORCE)
+
+    add_custom_command(
+      OUTPUT ${JSONITER_JAR}
+      COMMAND git clone -b ${JSONITER_VER} ${JSONITER_REPO_URL} ${JSONITER_REPO_DIR}
+      COMMAND mvn -Dmaven.repo.local=${CMAKE_CURRENT_BINARY_DIR}/.m2
+                  -Dmaven.test.skip=true -q package
+                  -f ${JSONITER_REPO_DIR}
+      )
+
+    add_custom_target(jsoniter
+      DEPENDS ${JSONITER_JAR}
+      )
+  endif()
+
+  # Build javaruntime jar
   add_jar(javaruntime
     src/com/ibm/onnxmlir/OMModel.java
     src/com/ibm/onnxmlir/OMLogger.java
+    src/com/ibm/onnxmlir/OMRunner.java
     src/com/ibm/onnxmlir/OMTensorList.java
     src/com/ibm/onnxmlir/OMTensor.java
+
+    INCLUDE_JARS ${JSONITER_JAR}
     )
+  if(TARGET jsoniter)
+    add_dependencies(javaruntime jsoniter)
+  endif()
   install_jar(javaruntime DESTINATION lib)
 
-  # ONNX_MLIR_LIBRARY_PATH is a generator expression which is not supported by add_jar as the output
-  # directory. Instead, we let add_jar place the jar file in the default location and copy it to the
-  # library path after it is built.
+  # ONNX_MLIR_LIBRARY_PATH is a generator expression which is not supported by
+  # add_jar as the output directory. Instead, we let add_jar place the jar file
+  # in the default location and copy it to the library path after it is built.
   add_custom_command(TARGET javaruntime POST_BUILD
-    COMMAND ${CMAKE_COMMAND} -E copy_if_different $<TARGET_PROPERTY:javaruntime,JAR_FILE> ${ONNX_MLIR_LIBRARY_PATH}/javaruntime.jar)
+    COMMAND ${CMAKE_COMMAND} -E copy_if_different
+            $<TARGET_PROPERTY:javaruntime,JAR_FILE> ${ONNX_MLIR_LIBRARY_PATH})
```

Comment:
If I remember correctly, we had to setup the full path in the copy command because otherwise the copy did not work correctly. After your change, is it still copying it to the correct file name?

Reply:
Yes the file is copied correctly.

Reply:
Ok, I tested this and on Windows, this will copy the jar file to ${ONNX_MLIR_LIBRARY_PATH} making that the filename if the directory does not exist already (a.k.a. the filename will be lib instead of being in the lib directory). Please keep the name in the copy command.